gcloud/storage/batch.py

```diff
@@ -0,0 +1,208 @@
+# Copyright 2014 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Batch updates / deletes of storage buckets / blobs.
+
+See: https://cloud.google.com/storage/docs/json_api/v1/how-tos/batch
+"""
+from email.encoders import encode_noop
+from email.generator import Generator
+from email.mime.application import MIMEApplication
+from email.mime.multipart import MIMEMultipart
+from email.parser import Parser
+import io
+import json
+import sys
+
+import six
+
+from gcloud._localstack import _LocalStack
+
+
+_BATCHES = _LocalStack()
+
+_PROXIED_ATTRS = [
```

Comment:
You can go for a fully-general proxy, a la [zope.proxy](http://zopeproxy.readthedocs.org/en/latest/narr.html), but the extra complexity isn't worth the trouble IMO.

The proxy could [define `__getattr__`, and delegate to the proxied object](https://github.com/zopefoundation/zope.proxy/blob/master/src/zope/proxy/__init__.py#L112).

In the end, the only "smell" to me is the whitelist of proxied attributes / methods;  other than that, copying the bound methods into the proxy seems very straightforward, compared to hooking `__getattr__` (or worse, [`__getattribute__`](https://github.com/zopefoundation/zope.proxy/blob/master/src/zope/proxy/__init__.py#L94)).